### ISSUE TITLE
Blank lines after import statements

### ISSUE THREAD

**Issue by victor-torres** [scrapy/scrapy#4654]
This issue was originated from a [discussion](https://github.com/scrapy/scrapy/pull/4434/files#r446676616) on pull request #4434.

## External resources

This is what the [Blank Lines](https://www.python.org/dev/peps/pep-0008/#blank-lines) section from the PEP 8 proposes:

> Surround top-level function and class definitions with two blank lines.
>
> Method definitions inside a class are surrounded by a single blank line.
>
> Extra blank lines may be used (sparingly) to separate groups of related functions. Blank lines may be omitted between a bunch of related one-liners (e.g. a set of dummy implementations).
> 
> Use blank lines in functions, sparingly, to indicate logical sections.
>
> Python accepts the control-L (i.e. ^L) form feed character as whitespace; Many tools treat these characters as page separators, so you may use them to separate pages of related sections of your file. Note, some editors and web-based code viewers may not recognize control-L as a form feed and will show another glyph in its place.

In the same line as PEP 8, [Google's Python Style Guide](https://google.github.io/styleguide/pyguide.html#35-blank-lines) also only mentions top-level function and class definitions:

> Two blank lines between top-level definitions, be they function or class definitions. One blank line between method definitions and between the class line and the first method. No blank line following a def line. Use single blank lines as you judge appropriate within functions or methods.

## Our usage

Let's take the `logger` variable definition, something very common across Python projects, as an example.

In Scrapy, we have some differences (both valid according to PEP 8):

- it's surrounded by two blank lines in [scrapy/downloadermiddlewares/cookies.py](https://github.com/scrapy/scrapy/blob/1d2b692666f9f5265e4845f8a30f04089515f7c5/scrapy/downloadermiddlewares/cookies.py) and in [scrapy/utils/signal.py](https://github.com/scrapy/scrapy/blob/1d2b692666f9f5265e4845f8a30f04089515f7c5/scrapy/utils/signal.py);
- but it's surrounded by only one blank line in [scrapy/spidermiddlewares/urllength.py](https://github.com/scrapy/scrapy/blob/dfbe1d95071acfbba159cea051530749b1684460/scrapy/spidermiddlewares/urllength.py) and in [scrapy/core/engine.py](https://github.com/scrapy/scrapy/blob/3d58e5e3be9e266b9b0f89b094823b76511a21b1/scrapy/core/engine.py)

Apparently, [Django is using only one blank line in this case](https://github.com/django/django/search?q=logger+%3D&unscoped_q=logger+%3D).

## Bottom line

What I'm trying to say is that there's no reference to this convention on the [Scrapy's Coding Style](https://docs.scrapy.org/en/master/contributing.html?highlight=blank%20lines#coding-style), and IMO it's quite subjective. 

**TL; DR;**

If this is important for the project to the point that we need to enforce it during code reviews, **I propose we implement automatic code checks with a linter or at least document it in the Scrapy's Coding Style for future reference**.

What do you think?

**Comment by kajoomba99** [comment]
Hello, I would like to solve this one

**Comment by abc516** [comment]
Hi @Gallaecio @victor-torres is this issue still open / ready for work?

**Comment by Gallaecio** [comment]
This is likely going to be addressed by https://github.com/scrapy/scrapy/issues/4658, which will be handled by core developers.
